src/engine/TransitivePathImpl.h

```diff
@@ -319,24 +319,39 @@ class TransitivePathImpl : public TransitivePathBase {
    * @return cppcoro::generator<TableColumnWithVocab> An generator for
    * the transitive hull computation
    */
-  static cppcoro::generator<TableColumnWithVocab> setupNodes(
+  static ad_utility::InputRangeTypeErased<TableColumnWithVocab> setupNodes(
       const TransitivePathSide& startSide,
       std::shared_ptr<const Result> startSideResult) {
     if (startSideResult->isFullyMaterialized()) {
-      // Bound -> var|id
-      ql::span<const Id> startNodes = startSideResult->idTable().getColumn(
-          startSide.treeAndCol_.value().second);
-      co_yield TableColumnWithVocab{&startSideResult->idTable(), startNodes,
-                                    startSideResult->getCopyOfLocalVocab()};
-    } else {
-      for (auto& [idTable, localVocab] : startSideResult->idTables()) {
+      auto getter = [&startSide, startSideResult]() {
         // Bound -> var|id
-        ql::span<const Id> startNodes =
-            idTable.getColumn(startSide.treeAndCol_.value().second);
-        co_yield TableColumnWithVocab{&idTable, startNodes,
-                                      std::move(localVocab)};
-      }
+        ql::span<const Id> startNodes = startSideResult->idTable().getColumn(
+            startSide.treeAndCol_.value().second);
```

Comment:
This code is duplicated below (get the startNodes given an idTable), make it a lambda that abstracts this away, then you can directly write `getStartNodes(startSideResult->idTable())` 
And reuse it in the case below (where you need it for every single table).